core/glrender/src/main/java/edu/cmu/cs/dennisc/render/gl/imp/adapters/GlrBox.java

```diff
@@ -55,184 +55,148 @@
 import edu.cmu.cs.dennisc.render.gl.imp.RenderContext;
 import edu.cmu.cs.dennisc.scenegraph.Box;
 
+import java.util.HashMap;
+import java.util.Map;
+
 /**
  * @author Dennis Cosgrove
- * Texture is defined by glTexCoord2f, which uses two value to crop the original picture
+ * Texture is defined by glTexCoord2f, which uses two values to crop the original picture
  */
 public class GlrBox extends GlrShape<Box> {
   private void glBox(Context c, boolean isLightingEnabled, boolean isSubElementRequired, boolean isTexturingEnabled) {
     //todo: revist vertex ordering
-    //xMin face
-    //c.gl.glColor3d( 1,1,1 );
     int id = 0;
+    int numberOfCorner = 4;
+    float[][] leftTexCoord = {{.75f, .66f}, {.75f, .33f}, {.5f, .33f}, {.5f, .66f}};
+    float[][] rightTexCoord = {{.25f, .66f}, {.25f, .33f}, {.0f, .33f}, {.0f, .66f}};
+    float[][] bottomTexCoord = {{.5f, .66f}, {.25f, .66f}, {.25f, 1.0f}, {.5f, 1.0f}};
+    float[][] topTexCoord = {{.5f, .0f}, {.25f, .0f}, {.25f, .33f}, {.5f, .33f}};
+    float[][] frontTexCoord = {{.5f, .33f}, {.25f, .33f}, {.25f, .66f}, {.5f, .66f}};
+    float[][] backTexCoord = {{.75f, .66f}, {1.0f, .66f}, {1.0f, .33f}, {.75f, .33f}};
+    double[][] leftVertex = {{this.xMin, this.yMin, this.zMax}, {this.xMin, this.yMax, this.zMax}, {this.xMin, this.yMax, this.zMin}, {this.xMin, this.yMin, this.zMin}};
+    double[][] rightVertex = {{this.xMax, this.yMin, this.zMin}, {this.xMax, this.yMax, this.zMin}, {this.xMax, this.yMax, this.zMax}, {this.xMax, this.yMin, this.zMax}};
+    double[][] bottomVertex = {{this.xMin, this.yMin, this.zMin}, {this.xMax, this.yMin, this.zMin}, {this.xMax, this.yMin, this.zMax}, {this.xMin, this.yMin, this.zMax}};
+    double[][] topVertex = {{this.xMin, this.yMax, this.zMax}, {this.xMax, this.yMax, this.zMax}, {this.xMax, this.yMax, this.zMin}, {this.xMin, this.yMax, this.zMin}};
+    double[][] frontVertex = {{this.xMin, this.yMax, this.zMin}, {this.xMax, this.yMax, this.zMin}, {this.xMax, this.yMin, this.zMin}, {this.xMin, this.yMin, this.zMin}};
+    double[][] backVertex = {{this.xMin, this.yMin, this.zMax}, {this.xMax, this.yMin, this.zMax}, {this.xMax, this.yMax, this.zMax}, {this.xMin, this.yMax, this.zMax}};
+    Map<String, float[][]> texCoordMap = new HashMap<>();
+    texCoordMap.put("left", leftTexCoord);
+    texCoordMap.put("right", rightTexCoord);
+    texCoordMap.put("bottom", bottomTexCoord);
+    texCoordMap.put("top", topTexCoord);
+    texCoordMap.put("front", frontTexCoord);
+    texCoordMap.put("back", backTexCoord);
+    Map<String, double[][]> vertexMap = new HashMap<>();
+    vertexMap.put("left", leftVertex);
+    vertexMap.put("right", rightVertex);
+    vertexMap.put("bottom", bottomVertex);
+    vertexMap.put("top", topVertex);
+    vertexMap.put("front", frontVertex);
+    vertexMap.put("back", backVertex);
 
     c.gl.glBegin(GL_QUADS);
+
+    // xMin face
+    // c.gl.glColor3d( 1,1,1 );
+    float[][] leftSideTex = texCoordMap.get("left");
```

Comment:
The next twelve lines are the same for all eight faces, except for the variable names and the normal for lighting.
You've gone this far, you could extract a method and eliminate a lot of repetition.